src/mr_reduction/filter_events.py

```diff
@@ -365,77 +326,263 @@ def filter_cross_sections(
+    """
+    ws_base_name = os.path.basename(file_path)
+    cross_sections: List[EventWorkspace] = list()
+
+    for entry in ["Off_Off", "On_Off", "Off_On", "On_On"]:
+        try:
+            ws_name = f"{ws_base_name}_{entry}"
+            ws = LoadEventNexus(Filename=file_path, NXentryName=f"entry-{entry}", OutputWorkspace=ws_name)
+            AddSampleLog(Workspace=ws, LogName="cross_section_id", LogText=entry)
+            cross_sections.append(ws_name)
+        except Exception as e:  # noqa E722
+            logger.information(f"Could not load entry {entry} from legacy data file: {e}")
+
+    return GroupWorkspaces(InputWorkspaces=cross_sections, OutputWorkspace=output_workspace)
+
+
+def get_workspace(
+    input_workspace: Optional[MantidWorkspace] = None, file_path: Optional[str] = None
+) -> MantidWorkspace:
+    """
+    Retrieve a Mantid workspace from either an existing workspace or by loading a file.
+
+    Parameters
+    ----------
     input_workspace : MantidWorkspace, optional
-        An existing Mantid workspace to process. If both `file_path` and `input_workspace` are provided,
-        `input_workspace` takes precedence.
-    pv_polarizer_state : str, optional
-        Name of the sample log for the polarizer state. Default is "SF1".
-    pv_analyzer_state : str, optional
-        Name of the sample log for the analyzer state. Default is "SF2".
-    pv_polarizer_veto : str, optional
-        Name of the sample log for the polarizer veto. Default is "SF1_Veto".
-    pv_analyzer_veto : str, optional
-        Name of the sample log for the analyzer veto. Default is "SF2_Veto".
-    check_devices : bool, optional
-        Whether to check for the presence of polarizer and analyzer devices. Default is True.
+        An existing Mantid workspace to process. If provided, this workspace will be used.
+    file_path : str, optional
+        Path to the Nexus file. If provided and `input_workspace` is None, the file will be loaded.
 
     Returns
     -------
-    WorkspaceGroup
-        A Mantid workspace group containing the filtered cross-sections.
+    MantidWorkspace
+        The retrieved Mantid workspace.
 
     Raises
     ------
     ValueError
-        If neither `file_path` nor `input_workspace` is provided.
-    RuntimeError
-        If no events remain after filtering.
+        If neither `input_workspace` nor `file_path` is provided.
+    """
+    if input_workspace is not None:
+        return workspace_handle(input_workspace)
 
-    Notes
-    -----
-    - If the file path ends with `.nxs`, it is assumed to be legacy data, and cross-sections are loaded independently.
-    - If no polarizer or analyzer information is available, the raw workspace is returned without filtering.
+    if file_path is not None:
+        return LoadEventNexus(Filename=file_path, OutputWorkspace="raw_events")
+
+    raise ValueError("Either 'file_path' or 'input_workspace' must be provided")
+
+
+def get_xs_list(
```

Comment:
I think the old name `split_events` is more descriptive (we can change `get_xs_list` in QuickNXS as well). Alternatively, we can change the name completely, maybe `split_events_into_cross_sections`?